src/base/IoManip.h

```diff
@@ -76,27 +76,31 @@ operator <<(std::ostream &os, const RawPointerT<Pointer> &pd)
 }
 
 /// std::ostream manipulator to print integers as hex numbers prefixed by 0x
+/// Normally used through the asHex() convenience function
 template <class Integer>
 class AsHex
 {
 public:
-    explicit AsHex(const Integer n): io_manip(n) {}
+    explicit AsHex(const Integer n, const size_t w=0) : io_manip(n), width(w) {}
     Integer io_manip; ///< the integer to print
+    size_t width; ///< the width to use number of digits to print
 };
 
 template <class Integer>
 inline std::ostream &
 operator <<(std::ostream &os, const AsHex<Integer> number)
 {
     const auto oldFlags = os.flags();
-    os << std::hex << std::showbase << number.io_manip;
+    os << "0x" << std::hex <<
+        std::setfill('0') << std::setw(number.width) <<
+        number.io_manip;
     os.setf(oldFlags);
```

Comment:
Please remember and restore the fill character that the new code is changing. It is not a flag. See how PrintHex() does it.

Please check whether width is automatically reset after the number is printed. If it is not, we need to remember and restore it as well.

Reply:
Added fill character reset.

Regarding resetting width, it's guaranteed to happen every time are sent to the stream (among other cases) " arithmetic types or void pointers (at Stage 3 of num_put::put())".

The fun thing is, I can see nothing in this code that constrains it to be used with integer types only. The width is reset in other cases, such as when char, c-string and basic_string are output, so there could be a scenario where some one misuses this code to output something which is not a number and doesn't get reduced to a c-string in its operator<<.

I'll still add a setw(0) because it gives us a hard guarantee at a probably low cost, but it could be debaeable